src/register/components/EmbeddableRegistrationPage.jsx

```diff
@@ -0,0 +1,261 @@
+import React, {
+  useEffect, useMemo, useState,
+} from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { getConfig } from '@edx/frontend-platform';
+import { sendPageEvent, sendTrackEvent } from '@edx/frontend-platform/analytics';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Form, StatefulButton } from '@edx/paragon';
+import { Helmet } from 'react-helmet';
+
+import ConfigurableRegistrationForm from './ConfigurableRegistrationForm';
+import RegistrationFailure from './RegistrationFailure';
+import { PasswordField } from '../../common-components';
+import { getThirdPartyAuthContext as getRegistrationDataFromBackend } from '../../common-components/data/actions';
+import { REDIRECT } from '../../data/constants';
+import { getAllPossibleQueryParams, setCookie } from '../../data/utils';
+import { clearRegistrationBackendError, registerNewUser } from '../data/actions';
+import { FORM_SUBMISSION_ERROR } from '../data/constants';
+import { getBackendValidations, isFormValid, prepareRegistrationPayload } from '../data/utils';
+import messages from '../messages';
+import { EmailField, NameField, UsernameField } from '../RegistrationFields';
+
+/**
+ * Main Registration Page component
+ */
+const EmbeddableRegistrationPage = () => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+
+  const flags = {
+    showConfigurableEdxFields: getConfig().SHOW_CONFIGURABLE_EDX_FIELDS,
+    showConfigurableRegistrationFields: getConfig().ENABLE_DYNAMIC_REGISTRATION_FIELDS,
+    showMarketingEmailOptInCheckbox: getConfig().MARKETING_EMAILS_OPT_IN,
+  };
+
+  const {
+    registrationFormData: backedUpFormData,
+    registrationError,
+    registrationError: {
```

Comment:
why are we using `registrationError` twice?

Reply:
@mubbsharanwar its nested destructuring, its mostly used when we both need the object and property from inside the object. So we destructure it like this.